src/fiatconnect/index.ts

```diff
@@ -142,7 +144,8 @@ export async function getFiatConnectQuotes(
   const quoteParams: CreateQuoteParams = {
     fiatType,
     cryptoType,
-    cryptoAmount: cryptoAmount.toString(),
+    ...(isUserInputCrypto(flow, digitalAsset) && { cryptoAmount: cryptoAmount.toString() }),
+    ...(!isUserInputCrypto(flow, digitalAsset) && { fiatAmount: fiatAmount.toString() }),
```

Comment:
```suggestion
    ...(isUserInputCrypto(flow, digitalAsset) ? { cryptoAmount: cryptoAmount.toString() } :  { fiatAmount: fiatAmount.toString() }),
```
code golf (more DRY)

Reply:
probly not formatted right but you get the idea